File: binomial_coeff.rb
# http://rosettacode.org/wiki/Evaluate_binomial_coefficients

class Fixnum
  
  def factorial
    (1..self).to_a.inject(:*)
  end

end

def binomial_coefficient(n,k)
  n.factorial / ((n - k).factorial * k.factorial)
end

p 5.factorial

p binomial_coefficient(5,3)

File: binomial_coeff.swift
#!/usr/bin/env swift
// http://rosettacode.org/wiki/Evaluate_binomial_coefficients

func factorial(n: Int) -> Int {
  return Array(1...n).reduce(1) { $0 * $1 }
}

func binomialCoefficient(n: Int, k: Int) -> Int {
  let numerator = factorial(n)
  let denominator = factorial(n - k) * factorial(k)
  return numerator/denominator
}

println(binomialCoefficient(5,3))
